servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +454,83 @@ def test_query_jobs_submission_pagination(self):
+                counts1Dict[status_count.status] = status_count.count
+
+            counts2Dict = {}
+            for status_count in counts2.counts:
+                counts2Dict[status_count.status] = status_count.count
+
+            return counts1Dict == counts2Dict
+
+        def test_get_job_aggregations(self):
+            name = flask.current_app.config['AGGREGATION_FILTER_LABEL']
+
+            labels = {
+                'aggregation-testing-unique-1': 'testing-rocks',
+                'aggregation-testing-unique-2': 'debugging-bad',
+                'aggregation-testing-unique-3': 'whatever-additional'
+            }
+
+            # Create an aborted job
+            job_aborted = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            time.sleep(10)
+            self.wait_status(job_aborted, ApiStatus.RUNNING)
+            self.must_abort_job(job_aborted)
+            self.wait_status(job_aborted, ApiStatus.ABORTED)
+
+            # Create a failed job
+            job_failed = self.api_job_id(
+                self.start_job('not_a_command', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.FAILED)
+
+            # Created a succeeded job
+            job_failed = self.api_job_id(
+                self.start_job('echo SUCCEEDED', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.SUCCEEDED)
+
+            # Created a running job
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            status_counts = StatusCounts.from_dict({
+                'counts': [{
+                    'status': u'Running',
+                    'count': 1
+                }, {
+                    'status': u'Failed',
+                    'count': 1
+                }, {
+                    'status': u'Succeeded',
+                    'count': 1
+                }, {
+                    'status': u'Aborted',
+                    'count': 1
+                }]
+            })
+
+            aggregation_resp = self.must_get_job_aggregations('HOURS_1')
+
+            self.assertEqual(
+                self.status_counts_equal(aggregation_resp.summary,
+                                         status_counts), True)
+
+            for aggregation in aggregation_resp.aggregations:
+                for entry in aggregation.entries:
+                    self.assertEqual(
+                        self.status_counts_equal(entry.status_counts,
+                                                 status_counts), True)
+
+            # Should have labels above presented in the aggregation response
+            labels = [
+                aggregation.key
+                for aggregation in aggregation_resp.aggregations
+            ]
+            self.assertEqual('aggregation-testing-unique-1' in labels, True)
```

Comment:
Should list out the full list of expected keys here, and then do `assertItemsEqual()`. Currently there might be extra stuff returned.